crates/primitives/src/l1/btc.rs

```diff
@@ -739,9 +749,24 @@ impl XOnlyPk {
             network,
         ))
     }
+
+    /// Converts [`XOnlyPk`] to [`Descriptor`].
+    pub fn to_descriptor(self) -> Descriptor {
+        let type_tag = DescriptorType::P2tr.to_u8();
+        Descriptor::from_vec([&[type_tag], self.0.as_bytes()].concat()).expect("infallible")
+    }
```

Comment:
Surely we can do this without allocating.  Is there not a `Descriptor` constructor that takes the xonly pubkey as an arg directly?

Reply:
Blocked by https://github.com/alpenlabs/bitcoin-bosd/pull/49

Reply:
Done in f1bd598db7187b072d6dd784a5ba6253154521f9